File: src/token.zig
pub const TokenType = enum {
    // Paw 核心关键字 (19个) - 极简设计
    keyword_fn,       // 函数定义
    keyword_let,      // 变量声明
    keyword_type,     // 类型定义（统一 struct/enum/trait）
    keyword_import,   // 导入
    keyword_pub,      // 可见性
    keyword_if,       // 条件
    keyword_else,     // 条件分支
    keyword_loop,     // 循环（统一 while/for）
    keyword_break,    // 中断
    keyword_return,   // 返回
    keyword_is,       // 模式匹配（统一 match）
    keyword_as,       // 类型转换
    keyword_async,    // 异步
    keyword_await,    // 等待
    keyword_self,     // 实例引用
    keyword_Self,     // 类型引用
    keyword_mut,      // 可变性
    keyword_true,     // 布尔真
    keyword_false,    // 布尔假
    keyword_in,       // 上下文关键字（用于 loop for）

    // 内置类型（Rust 风格，无别名）
    // 整数类型（有符号）
    type_i8,      // 8位有符号整数
    type_i16,     // 16位有符号整数
    type_i32,     // 32位有符号整数（默认）
    type_i64,     // 64位有符号整数
    type_i128,    // 128位有符号整数
    
    // 整数类型（无符号）
    type_u8,      // 8位无符号整数
    type_u16,     // 16位无符号整数
    type_u32,     // 32位无符号整数
    type_u64,     // 64位无符号整数
    type_u128,    // 128位无符号整数
    
    // 浮点类型
    type_f32,     // 32位浮点数
    type_f64,     // 64位浮点数（默认）
    
    // 其他类型
    type_bool,    // 布尔类型
    type_char,    // 字符类型
    type_string,  // 字符串类型
    type_void,    // 空类型

    // 标识符和字面量
    identifier,
    int_literal,
    float_literal,
    string_literal,
    char_literal,

    // 运算符
    plus,        // +
    minus,       // -
    star,        // *
    slash,       // /
    percent,     // %
    
    eq,          // ==
    ne,          // !=
    lt,          // <
    le,          // <=
    gt,          // >
    ge,          // >=
    
    and_and,     // &&
    or_or,       // ||
    bang,        // !
    question,    // ? (错误传播)
    
    assign,      // =
    // 🆕 复合赋值操作符
    add_assign,  // +=
    sub_assign,  // -=
    mul_assign,  // *=
    div_assign,  // /=
    mod_assign,  // %=
    arrow,       // ->
    fat_arrow,   // =>
    
    // 分隔符
    lparen,      // (
    rparen,      // )
    lbrace,      // {
    rbrace,      // }
    lbracket,    // [
    rbracket,    // ]
    
    comma,       // ,
    semicolon,   // ;
    colon,       // :
    double_colon, // ::
    dot,         // .
    dot_dot,     // .. (范围，不包含结束)
    dot_dot_eq,  // ..= (范围，包含结束)
    
    // 特殊
    eof,
    invalid,
};

pub const Token = struct {
    type: TokenType,
    lexeme: []const u8,
    line: usize,
    column: usize,

    pub fn init(token_type: TokenType, lexeme: []const u8, line: usize, column: usize) Token {
        return Token{
            .type = token_type,
            .lexeme = lexeme,
            .line = line,
            .column = column,
        };
    }
};

File: src/frontend/lexer/token.cpp
//===--- token.cpp - Token Implementation ------------------------*- C++ -*-===//

#include "token.h"

namespace pawc {

const char* tokenTypeToString(TokenType type) {
    switch (type) {
        case TokenType::END_OF_FILE: return "END_OF_FILE";
        case TokenType::INVALID: return "INVALID";
        case TokenType::IDENTIFIER: return "IDENTIFIER";
        case TokenType::INT_LITERAL: return "INT_LITERAL";
        case TokenType::FLOAT_LITERAL: return "FLOAT_LITERAL";
        case TokenType::STRING_LITERAL: return "STRING_LITERAL";
        case TokenType::CHAR_LITERAL: return "CHAR_LITERAL";
        
        case TokenType::AS: return "AS";
        case TokenType::BREAK: return "BREAK";
        case TokenType::CONTINUE: return "CONTINUE";
        case TokenType::ELSE: return "ELSE";
        case TokenType::ENUM: return "ENUM";
        case TokenType::FALSE: return "FALSE";
        case TokenType::FN: return "FN";
        case TokenType::IF: return "IF";
        case TokenType::IMPORT: return "IMPORT";
        case TokenType::IN: return "IN";
        case TokenType::INTERFACE: return "INTERFACE";
        case TokenType::IS: return "IS";
        case TokenType::LET: return "LET";
        case TokenType::LOOP: return "LOOP";
        case TokenType::NULL_KW: return "NULL_KW (deprecated)";
        case TokenType::OK: return "OK";
        case TokenType::ERR: return "ERR";
        case TokenType::PUB: return "PUB";
        case TokenType::RETURN: return "RETURN";
        case TokenType::SELF_LOWER: return "SELF_LOWER";
        case TokenType::SELF_UPPER: return "SELF_UPPER";
        case TokenType::STRUCT: return "STRUCT";
        case TokenType::SUPPORT: return "SUPPORT";
        case TokenType::TRUE: return "TRUE";
        case TokenType::TYPE: return "TYPE";
        case TokenType::WHERE: return "WHERE";
        case TokenType::WITH: return "WITH";
        
        case TokenType::PLUS: return "PLUS";
        case TokenType::MINUS: return "MINUS";
        case TokenType::STAR: return "STAR";
        case TokenType::SLASH: return "SLASH";
        case TokenType::PERCENT: return "PERCENT";
        case TokenType::AMP: return "AMP";
        case TokenType::PIPE: return "PIPE";
        case TokenType::CARET: return "CARET";
        case TokenType::TILDE: return "TILDE";
        case TokenType::BANG: return "BANG";
        case TokenType::QUESTION: return "QUESTION";
        case TokenType::EQ: return "EQ";
        case TokenType::LT: return "LT";
        case TokenType::GT: return "GT";
        case TokenType::DOT: return "DOT";
        case TokenType::COMMA: return "COMMA";
        case TokenType::COLON: return "COLON";
        case TokenType::SEMICOLON: return "SEMICOLON";
        case TokenType::AT: return "AT";
        
        case TokenType::PLUS_EQ: return "PLUS_EQ";
        case TokenType::MINUS_EQ: return "MINUS_EQ";
        case TokenType::STAR_EQ: return "STAR_EQ";
        case TokenType::SLASH_EQ: return "SLASH_EQ";
        case TokenType::PERCENT_EQ: return "PERCENT_EQ";
        case TokenType::AMP_EQ: return "AMP_EQ";
        case TokenType::PIPE_EQ: return "PIPE_EQ";
        case TokenType::CARET_EQ: return "CARET_EQ";
        case TokenType::EQ_EQ: return "EQ_EQ";
        case TokenType::NOT_EQ: return "NOT_EQ";
        case TokenType::LESS_EQ: return "LESS_EQ";
        case TokenType::GREATER_EQ: return "GREATER_EQ";
        case TokenType::AND_AND: return "AND_AND";
        case TokenType::OR_OR: return "OR_OR";
        case TokenType::LT_LT: return "LT_LT";
        case TokenType::GT_GT: return "GT_GT";
        case TokenType::COLON_COLON: return "COLON_COLON";
        case TokenType::DOT_DOT: return "DOT_DOT";
        case TokenType::ARROW: return "ARROW";
        case TokenType::FAT_ARROW: return "FAT_ARROW";
        
        case TokenType::LPAREN: return "LPAREN";
        case TokenType::RPAREN: return "RPAREN";
        case TokenType::LBRACE: return "LBRACE";
        case TokenType::RBRACE: return "RBRACE";
        case TokenType::LBRACKET: return "LBRACKET";
        case TokenType::RBRACKET: return "RBRACKET";
        
        default: return "UNKNOWN";
    }
}

std::string Token::toString() const {
    return std::string(tokenTypeToString(type)) + "(" + lexeme + ")";
}

} // namespace pawc

